packages/wallets/src/wallets/wallet-factory.ts

```diff
@@ -290,19 +371,22 @@ export class WalletFactory {
             );
         }
 
-        // Check that each input delegated signer exists in the wallet
-        // (wallet can have additional signers that weren't specified in input)
-        for (const argSigner of inputDelegatedSigners) {
+        for (const inputSigner of inputDelegatedSigners) {
             const matchingExistingSigner = existingDelegatedSigners.find(
-                (existingSigner) => existingSigner.locator === argSigner.signer
+                (existingSigner) => existingSigner.type === inputSigner.type
```

Comment:
can we also use locators here?

Reply:
we only need to compare the information that we do have. so a user can get a wallet and if it's going to act as a delegated signer, we verify that delegated signer matches one in the wallet, we can ignore the rest